llvm/lib/IR/Verifier.cpp

```diff
@@ -4378,6 +4378,29 @@ void Verifier::visitEHPadPredecessors(Instruction &I) {
   }
 }
 
+// Recursively extract types from the LandingPad type
+// Based on ComputeValueVTs
+static void linearizeLandingPadTypes(Type *Ty,
```

Comment:
This is effectively a copy of `ComputeValueVTs`, but without type conversion and offset calculation which depends on `TargetLowering information` and `DataLayout`.
Not sure where to put this code or whether there is some existing functionality I can reuse.

Better suggestions are highly appreciated!

Reply:
This function is overkill for what you're verifying. You can just check ArrayType::getNumElements or StructType::getNumElements is 2?